src/nix/develop.cc

```diff
@@ -697,6 +697,10 @@ struct CmdDevelop : Common, MixEnvironment
             }
         }
 
+        // Release our references to eval caches to ensure they are persisted to disk, because
+        // we are about to exec out of this process without running C++ destructors.
+        getEvalState()->evalCaches.clear();
```

Comment:
I guess we should also do this in `nix shell`, `nix run` etc.

Reply:
added to all commands.

i tried to come up with a good abstraction but couldn't come up with anything that didn't add a dependency from `libcmd` to `nix`, so i just went with the copypasta route. i also renamed `runProgramInStore` to `execProgramInStore` to better highlight the nature of the function.